### ISSUE TITLE
Difficulty Detecting Aborted Streaming Request on Nitro Side

### ISSUE THREAD

**Issue by XStarlink** [nitrojs/nitro#2118]
### Environment

Nitro: 2.8.1
Node: 18

### Reproduction

https://stackblitz.com/edit/nuxt-tip-cancel-fetch-request-ysnjhe?file=app.vue,components%2FFetchStream.vue

### Describe the bug

Hello Nitro Team,

Thanks for the incredible work you're doing with Nuxt and Nitro!

I'm currently facing a challenge, I'm trying to detect on the Nitro side when a request has been aborted or closed, specifically in the case of a streaming request. I've tried numerous approaches, but I haven't been successful in detecting a request being aborted on the Nitro side to halt the execution of the event handler and avoid further processing.

[I've created a small reproduction for you to take a look at here
](https://stackblitz.com/edit/nuxt-tip-cancel-fetch-request-ysnjhe?file=components%2FFetchStream.vue,server%2Fapi%2Fstream.ts)

I thank you very much for your help, and I wish you an excellent day!

**Comment by goetzrobin** [comment]
@XStarlink did you ever find a solution for this?

**Comment by XStarlink** [comment]
@goetzrobin Yes there was an issue in nitro, [look at this discussion ](https://gist.github.com/Atinux/05836469acca9649fa2b9e865df898a2) but now if you are in the latest Nuxt version, you can use this code in your server route:

```js
event.node.res.on('close', () => {
  console.log('Connection closed by client');
});
```

I've updated the Stackblitz example linked to this to show you, but apparently your project must be running on a node server to use the code snippet above otherwise it won't detect that the connection is closed.

For other environments, I don't know how to detect this, but if someone can answer this question, it would be very interesting.

**Comment by mlgarchery** [comment]
Thanks @XStarlink it happens that your code works in a Cloudflare environment.
For next readers that might be interested in aborting an AI stream an save tokens in your Cloudflare workers, you can use this (here I am using LangChain):
```
  const controller = new AbortController();
  event.node.res.on("close", () => {
    controller.abort();
  });
  const stream = await chatModel.stream(msgs, { signal: controller.signal });
  ```

**Comment by pi0x** [comment]
Fixed in v3 — handlers get a standard `event.req.signal` that aborts when the client goes away, so you can stop a stream with `event.req.signal.addEventListener("abort", ...)` or pass it straight to fetch.

_Triaged with AI assistance._
